docs/orchestrating-elastic-stack-applications/logstash.asciidoc

```diff
@@ -0,0 +1,569 @@
+            user => "${QS_ES_USER}"
+            password => "${QS_ES_PASSWORD}"
+            cacert => "${QS_ES_SSL_CERTIFICATE_AUTHORITY}"
+          }
+        }
+----
+
+Alternatively, it can be provided through a Secret specified in the `spec.pipelinesRef` element. The Secret must have a `logstash.yml` entry with this configuration:
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRefs:
+    - clusterName: qs
+      name: quickstart
+  pipelinesRef:
+    secretName: quickstart-pipeline
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: quickstart-pipeline
+stringData:
+  pipelines.yml: |-
+    - pipeline.id: main
+      config.string: |
+        input {
+          beats {
+            port => 5044
+          }
+        }
+        output {
+          elasticsearch {
+            hosts => [ "${QS_ES_HOSTS}" ]
+            user => "${QS_ES_USER}"
+            password => "${QS_ES_PASSWORD}"
+            cacert => "${QS_ES_SSL_CERTIFICATE_AUTHORITY}"
+          }
+        }
+
+----
+
+Logstash on ECK will* support all options present in `pipelines.yml`, including settings to update the number of workers, and
+ the size of the batch that the pipeline will process. This also includes using `path.config` to point to volumes
+ mounted on the logstash container:
+
+[source,yaml,subs="attributes,+macros,callouts"]
+----
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRefs:
+    - clusterName: qs
+      name: quickstart
+  pipelines:
+    - pipeline.id: main
+      config.string: |
+        input {
+          beats {
+            port => 5044
+          }
+        }
+        output {
+          elasticsearch {
+            hosts => [ "${QS_ES_HOSTS}" ]
+            user => "${QS_ES_USER}"
+            password => "${QS_ES_PASSWORD}"
+            cacert => "${QS_ES_SSL_CERTIFICATE_AUTHORITY}"
+          }
+        }
+----
+
+NOTE: Persistent Queues and Dead Letter Queues will be supported in a later release, but are not currently supported.
+
+
+[id="{p}-logstash-pipelines-es"]
+=== Using Elasticsearch in Logstash Pipelines
+
+The `spec.elasticsearchRefs` section provides a mechanism to help configure Logstash to estabish a secured connection to one or more managed Elasticsearch clusters. By default, it targets all nodes in your cluster. If you want to direct traffic to specific nodes of your Elasticsearch cluster, refer to <<{p}-traffic-splitting>> for more information and examples.
+
+In order to use `elasticsearchRefs` in a logstash pipeline, the logstash operator will create the necessary resources from the associated elasticsearch, and provide environment variables to allow these resources to be accessed from pipeline configuration, and will be replaced at runtime with the appropriate values.
```

Comment:
```suggestion
To use `elasticsearchRefs` in a Logstash pipeline, the Logstash operator will create the necessary resources from the associated Elasticsearch, and provides environment variables to allow these resources to be accessed from the pipeline configuration, and will be replaced at runtime with the appropriate values.
```
You might want to split this into smaller sentences. It's too long and a bit confusing.